src/dcd/server/autocomplete/ufcs.d

```diff
@@ -0,0 +1,171 @@
+module dcd.server.autocomplete.ufcs;
+
+import dcd.server.autocomplete.util;
+import dsymbol.symbol;
+import dsymbol.scope_;
+import dcd.common.messages;
+import std.functional : unaryFun;
+import std.algorithm;
+import std.array;
+import std.range;
+import dsymbol.builtin.names;
+import std.string;
+import dparse.lexer : tok;
+import std.regex;
+import containers.hashset : HashSet;
+
+void lookupUFCS(Scope* completionScope, DSymbol* beforeDotSymbol, size_t cursorPosition, ref AutocompleteResponse response)
+{
+    // UFCS completion
+    DSymbol*[] ufcsSymbols = getSymbolsForUFCS(completionScope, beforeDotSymbol, cursorPosition);
+    response.completions ~= map!(s => createCompletionForUFCS(s))(ufcsSymbols).array;
+}
+
+AutocompleteResponse.Completion createCompletionForUFCS(const DSymbol* symbol)
+{
+    return AutocompleteResponse.Completion(symbol.name, symbol.kind, "(UFCS) " ~ symbol.callTip, symbol
```

Comment:
how about we use symbol.callTip as-is, but add a `bool ufcs` (or some kind of uint flag with this as hint) and add the `(UFCS) ` prefix from dcd-client? I think it would be more clean and also this concat is otherwise an allocation every time.

Reply:
Sure we can do that, but I don't think the extra allocation really matters much, 
 so you essentially want a new completion kind?
example:

```
identifiers
parts	v
name	v
location	v
qualifier	v
kind	v
type	v
resolvedType	v
calltip	v
getPartByName	f
ufcsBar F
```
Where F means UFCS
I just choose F, since the letter  f is the completion kind for function, but whatever makes sense.

Reply:
yeah I think that'd be best - the UFCS functions can be displayed and processed slightly differently to normal functions, but still get displayed in the same completion list with the same usual syntax.

Reply:
I agree it's proably a better solution. I will take look it.